arch/arm/include/arch.h

```diff
@@ -164,6 +164,8 @@ typedef struct arch_addrenv_s arch_addrenv_t;
  * Public Function Prototypes
  ****************************************************************************/
 
+uintptr_t up_read_cpuid(void);
```

Comment:
why not reuse this_cpu() / up_cpu_index()?

Reply:
I think you don't need to add a new API, because your implementation is not common code, and the relevant interfaces have been implemented. You can refer to the implementation of arm_cpuinfo.c:

cortex-a/r:
cpuid = CP15_GET(MIDR);
cortex-m:
cpuid = getreg32(ARMV6M_SYSCON_CPUID);

Reply:
the interface desgin is wrong, since how to interpret the returned value is totally different on different arch.